### PR TITLE
feat: add modelscan scanner for binary model files

### PR DESCRIPTION
**Pull request by thypon**
Adds [modelscan](https://github.com/protectai/modelscan) as a new scanner for binary model files (pickle, PyTorch, NumPy, H5, Keras, SavedModel).

## Why

Replaces the closed PR #707 (fickling-based) which was blocked by reviewdog's inability to handle binary diffs. This bypasses reviewdog entirely and posts file-level review comments directly via the GitHub REST API with `subject_type: "file"`.

## What

- **`assets/modelscan-audit.py`** — Magic-byte detection dispatches to modelscan scanner APIs (`scan_pickle_bytes`, `scan_pytorch`, `scan_numpy`, H5/Keras/SavedModel scanners). Extensionless binary files still get scanned correctly via magic bytes.
- **`src/modelscanPostComments.js`** — Runs Python audit, parses JSON, deduplicates against existing review threads, posts file-level comments capped at 10/run. Comments include `<!-- modelscan -->` + `<!-- Category: security -->` markers so existing `commentsNumber`, `cleanupComments`, and label triggers work automatically.
- **`src/modelscanPostComments.test.js`** — 6 tests covering no findings, comment posting, dedup, outdated re-post, 10-cap, and 422 handling.
- **`actions/main/action.yml`** — New `modelscan_enabled` input (comma-separated list, default `"all"`). Supports `"pickle,numpy,pytorch"` for lightweight-only, `"false"` to disable entirely.
- **`actions/main/action.cjs`** — Modelscan step inserted after changed-files write, before codeowners step, gated by `modelscan_enabled`.
- **`requirements.txt`** — Added `modelscan~=0.1`, `numpy>=1.20`, `h5py>=3.0` (commented for slim-install), `tensorflow>=2.10` (commented).

## Scanner enable/disable matrix

| Want | `modelscan_enabled` input | requirements.txt |
|------|--------------------------|------------------|
| All scanners (default) | `"all"` | keep all extras |
| Lightweight only | `"pickle,numpy,pytorch"` | remove h5py, tensorflow |
| No tensorflow | `"pickle,numpy,pytorch,h5"` | remove tensorflow |
| Disable modelscan | `"false"` | (optional: remove all) |

## How it works in CI

1. PR opens -> `pullRequestChangedFiles.js` writes `all_changed_files.txt`
2. `modelscanPostComments` runs `modelscan-audit.py` which reads each file's magic bytes, dispatches to matching scanner
3. Each finding (severity LOW+) deduplicates against existing review threads with `<!-- modelscan -->` marker
4. File-level comment posted on the changed file's path
5. `commentsNumber.js` picks it up via `<br>Cc ` filter -> `needs-security-review` label triggers
6. `cleanupComments.js` auto-deletes stale comments on re-push

See [PR #707](https://github.com/brave/security-action/pull/707) for the reviewdog binary-file limitation analysis.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,15 @@
 pip-audit~=2.9.0
-setuptools~=79.0.1
+setuptools~=83.0.0
```
Comment (diracdeltas) [inline requirements.txt:2]: it would be good to lock dependencies here but i guess you'd have to switch to UV

Comment (mschfh) [reply requirements.txt:2]: +1 for uv, it also allows moving tensorflow to a separate dependency group.

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,15 @@
...
+# Heavyweight scanners need the extras below — to slim the install:
+#   1. Set modelscan_enabled action input to exclude h5/keras/saved_model
+#      (e.g. modelscan_enabled: 'pickle,numpy,pytorch')
+#   2. Remove the matching line(s) below.
+modelscan~=0.8
+numpy>=1.20
+# ↓ heavyweight (~10MB) — remove to drop h5 scanner ↓
+h5py>=3.0    # required by H5LambdaDetectScan (.h5)
+# ↓↓↓ heavyweight (~500MB) — remove to drop keras + saved_model scanners ↓↓↓
+tensorflow>=2.10    # required by KerasLambdaDetectScan (.keras) + SavedModelLambdaDetectScan (.pb)
```
Comment (mschfh) [inline requirements.txt:14]: A 500MB dependency would pollute the actions cache

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+[project]
+name = "security-action"
+version = "0.0.0"
+requires-python = ">=3.12,<3.13"
```
Comment (mschfh) [inline pyproject.toml:4]: 3.12 is EOL (although still receives security updates), current stable is 3.14

Comment (thypon) [reply pyproject.toml:4]: modelscan requires less than 3.13, so we are stuck there
